### PR TITLE
Defined the contact status given the contact force

### PR DESCRIPTION
**Pull request by cmastalli**

### CODE REVIEW

**Comment by wxmerkt** [comment]
Do we want to do this automatically in the interface? The explicit flag was to have the status from a probabilistic contact estimator since we may have measured forces but still have the status to be "open"/"not in contact". E.g. on Anymal when we may have up to 50 N phantom forces but the robot should not set the flag unless the probabilistic contact estimators says so

**Comment by wxmerkt** [comment]
^ the above said, we don't currently have an argument to explicitly pass the state which we do require during control. Should we pass the state as boolean explicitly (extend the argument list) and only if not passed use the norm of the force?

**Comment by cmastalli** [comment]
> Should we pass the state as boolean explicitly (extend the argument list) and only if not passed use the norm of the force?

It makes sense. What about using a default argument? If the user has not passed this boolean, then we set as proposed in this PR.

If you agree on it, I could implement it now.

**Comment by wxmerkt** [comment]
Happy with this option - let's also trigger a warning that it should be set or else it's assumed based on the norm

**Comment by cmastalli** [comment]
Actually, it might be better to create the python bindings, and remove this file.
What do you think?

If we follow the development flow, we should implement these bindings here: https://github.com/ori-drs/state_feedback_controller_interface

**Comment by wxmerkt** [comment]
In principle, Python bindings would be nice - however, passing serialized/deserialized ROS message types isn't easy (it's possible and we've done it before, but requires custom converters). Unless it's a bottleneck we may want to go the easy route and amend the Python module here and add the bindings as a nice-to-have for the future. What do you think?

**Comment by cmastalli** [comment]
> In principle, Python bindings would be nice - however, passing serialized/deserialized ROS message types isn't easy (it's possible and we've done it before, but requires custom converters). Unless it's a bottleneck we may want to go the easy route and amend the Python module here and add the bindings as a nice-to-have for the future. What do you think?

This shouldn't substitute the publishers/subscribers that you already have bind.
But, it is still useful to bind the `WholeBodyInterface`.
Indeed, I have some legacy (in Python) code that used it.

File: src/whole_body_state_conversions/whole_body_interface.py [hunk 1]
```diff
@@ -123,6 +123,8 @@ def writeToMessage(self, t, q, v=None, tau=None, p=dict(), pd=dict(), f=dict(),
                 contact_msg.wrench.torque.x = force.angular[0]
                 contact_msg.wrench.torque.y = force.angular[1]
                 contact_msg.wrench.torque.z = force.angular[2]
+                if np.linalg.norm(force.linear) > 0.:
```
Comment (wxmerkt) [inline src/whole_body_state_conversions/whole_body_interface.py]:
```suggestion
                if len(contact_info) == 3:
                    contact_msg.status = contact_info[2]
```

**Comment by wxmerkt** [comment]
For the MPC, I've added a C++ subscriber that avoids having to bind the WholeBodyStateInterface. I tried that but passing a  ROS message from Python to C++ is a challenge and hasn't worked so far with the complicated WholeBodyStateInterface. For the above change, we should pass the state in explicitly (as the writeFromMessage returns it), rather than inferring from the norm of forces.

File: src/whole_body_state_conversions/whole_body_interface.py (hunk 1 above)
Comment (cmastalli) [reply src/whole_body_state_conversions/whole_body_interface.py]: Done! I have added a deprecate message to keep the previous behaviour.

Note that I have a similar code in whole-body-state-msgs. And there is legacy code that uses.
I will remove the deprecation routine as soon as I update my code.

**Pull request merged**